pkg/k8s/namespaces/crud.go

```diff
@@ -21,3 +27,100 @@ func IsOktetoNamespace(ns string, c *kubernetes.Clientset) bool {
 	}
 	return n.Labels[OktetoLabel] == "true"
 }
+
+//CheckAvailableResources checks if therre are available resources for activating the dev env
+func CheckAvailableResources(dev *model.Dev, create bool, d *appsv1.Deployment, c *kubernetes.Clientset) error {
+	needCPU := resource.MustParse("0")
+	needMemory := resource.MustParse("0")
+	if !existsSyncthing(dev, c) {
+		needCPU.Add(resource.MustParse("500m"))
+		needMemory.Add(resource.MustParse("256Mi"))
+	}
+	lr, err := c.CoreV1().LimitRanges(dev.Namespace).Get(dev.Namespace, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	needCPU.Add(matchCPU(dev, true, 1, d.Spec.Template.Spec, lr))
+	needMemory.Add(matchMemory(dev, true, 1, d.Spec.Template.Spec, lr))
+	if !create {
+		needCPU.Sub(matchCPU(dev, false, d.Status.Replicas, d.Spec.Template.Spec, lr))
+		needMemory.Sub(matchMemory(dev, false, d.Status.Replicas, d.Spec.Template.Spec, lr))
+	}
+	rq, err := c.CoreV1().ResourceQuotas(dev.Namespace).Get(dev.Namespace, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	usedCPU := rq.Status.Used[apiv1.ResourceLimitsCPU]
+	usedCPU.Add(needCPU)
+	totalCPU := rq.Status.Hard[apiv1.ResourceLimitsCPU]
+	if usedCPU.Value() > totalCPU.Value() {
+		usedCPU.Sub(needCPU)
+		totalCPU.Sub(usedCPU)
+		return fmt.Errorf("Activating your dev environment requires %s cpu and your remaining quota is %s cpu", needCPU.String(), totalCPU.String())
```

Comment:
Do you think we should send a more generic error, and then have the UI display this as a `hint`? (the blue text)